build.gradle

```diff
@@ -1,151 +1,119 @@
-	apply from: 'private.gradle'
-}
-
-archivesBaseName = project.archives_base_name
-version = project.mod_version
-group = project.maven_group
+archivesBaseName = "cotton"
+version = "1.0.0-rc.1"
+group = "io.github.cottonmc.cotton"
 
-minecraft {
-	refmapName = 'mixins.cotton.refmap.json'
-}
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
 
 repositories {
-	mavenCentral()
-	maven { url "http://server.bbkr.space:8081/artifactory/libs-snapshot/" }
-	maven { url "http://server.bbkr.space:8081/artifactory/libs-release/" }
+    mavenLocal()
+    maven {
+        name = "Cotton"
+        url = "http://server.bbkr.space:8081/artifactory/libs-release/"
+    }
+    maven {
+        name = "Cotton (snapshots)"
+        url = "http://server.bbkr.space:8081/artifactory/libs-snapshot/"
+    }
 }
 
 dependencies {
-	//to change the versions see the gradle.properties file
-	minecraft "com.mojang:minecraft:${project.minecraft_version}"
-	mappings "net.fabricmc:yarn:${project.yarn_mappings}"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
-
-	// Fabric API. This is technically optional, but you probably want it anyway.
-	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
-
-	modApi "io.github.cottonmc:LibCD:${project.libcd_version}"
-	include "io.github.cottonmc:LibCD:${project.libcd_version}"
-
-	compileOnly ("com.google.code.findbugs:jsr305:3.0.2") { transitive = false }
-    // modApi jankson for properly including it as a dep with Loom
-	modApi ("blue.endless:jankson:1.1.2") //we want jankson available now that it's included instead of shadowed!
-	include ("blue.endless:jankson:1.1.2") { transitive = false }
-
-	testImplementation ("org.junit.jupiter:junit-jupiter-api:5.4.1") //{ transitive=false; };
-	testRuntimeOnly ("org.junit.jupiter:junit-jupiter-engine:5.4.1") //{ transitive=false; };
+    minecraft "com.mojang:minecraft:$minecraftVersion"
+    mappings "net.fabricmc:yarn:$yarnMappings"
+    modImplementation "net.fabricmc:fabric-loader:$loaderVersion"
+
+    // `modApi`s below are for the dev env
+    def modules = ["player-events", "logging", "config", "datapack", "commons", "cauldron"]
+    modules.each { module ->    
+        modApi ("io.github.cottonmc.cotton:cotton-$module:latest.release") { transitive = false }
+        include "io.github.cottonmc.cotton:cotton-$module:latest.release"
+    }
+    modApi ("io.github.cottonmc:Jankson:1.0.0+j1.1.2") { transitive = false }
+    modApi ("io.github.cottonmc:LibCD:1.3.0+1.14.3") { transitive = false }
+    include "io.github.cottonmc:Jankson:1.0.0+j1.1.2"
+    include "io.github.cottonmc:LibCD:1.3.0+1.14.3"
+
+    compileOnly ("com.google.code.findbugs:jsr305:3.0.2") { transitive = false }
+
+    // For ModMenu compatibility tests
+    modImplementation "net.fabricmc.fabric-api:fabric-api:0.3.0+build.187"
+    modImplementation "io.github.prospector.modmenu:ModMenu:1.6.3-94"
```

Comment:
This (or both, depending on if Fabric API is needed during compilation) can be `modRuntime`.

Reply:
Nope, they can't. Tested